ql/src/java/org/apache/hadoop/hive/ql/parse/ReplicationSemanticAnalyzer.java

```diff
@@ -378,14 +375,23 @@ private void analyzeReplLoad(ASTNode ast) throws SemanticException {
             LOG.debug("Updated taskChainTail from {}{} to {}{}",
                 taskChainTail.getClass(), taskChainTail.getId(), barrierTask.getClass(), barrierTask.getId());
             taskChainTail = barrierTask;
-            evstage++;
           }
         }
+
+        // If any event is there and db name is known, then dump the start and end logs
+        if (evTaskRoot != taskChainTail) {
```

Comment:
even though reference based check will work since they are objects it will be better to have  ! evTaskRoot.equals(taskChainTail)